connect_four_game.cpp

```diff
@@ -0,0 +1,164 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+// Below is the implementation of Connect Four game.
+// Follow up Question to be solved next- Algorithm to find the best colum to drop the coin.
+class ConnectFour
+{
+
+private:
+    int totalRows;
+    int totalCols;
+    // board configuration
+    vector<vector<int>> board;
+    //shows the top available spot for each column
+    vector<int> topSpot;
+
+public:
+    ConnectFour(int totalRows, int totalCols)
+    {
+        this->totalRows = totalRows;
+        this->totalCols = totalCols;
+        board.resize(totalRows, vector<int>(totalCols, 0));
+        //bottom spot available for each column
+        topSpot.resize(totalCols, totalRows - 1);
+    }
+
+    int move(int col, int player)
+    {
+        // This function returns
+        // -1 -> for invalid move
+        // 1 -> player doesn't win after this move
+        // int player -> if player wins after this move
+        // Time Complexity - O(1)
+
+        // If that column doesn't exist or that column is filled,
+        // it's an invalid move.
+        if((col < 0) || (col >= totalCols) || (topSpot[col] < 0))
+        {
+            cout << "Not a valid move\n";
+            return -1;
+        }
+        else
+        {
+            board[topSpot[col]][col] = player;
+            topSpot[col]--;
+        }
+
+        if(isWinner(col, player) != 0)
+            return player;
+
+        return 0;
+    }
+
+    bool isWinner(int col, int player)

```

Comment:
This method returns boolean. But in the return statements inside the method, the return values can be 0, 1 and 2. How would that work?